appinventor/appengine/src/com/google/appinventor/server/LoginServlet.java

```diff
@@ -440,8 +440,8 @@ private String trimPage(HttpServletRequest req) {
     return sb.toString();
   }
 
-  private void fail(HttpServletRequest req, HttpServletResponse resp, String error) throws IOException {
-    resp.sendRedirect("/login/?error=" + sanitizer.sanitize(error));
+  private void fail(HttpServletRequest req, HttpServletResponse resp, String error, String locale) throws IOException {
+    resp.sendRedirect("/login/?locale=" + locale + "&error=" + sanitizer.sanitize(error));
```

Comment:
Because `locale` is coming from the client, you should call `sanitizer.sanitize` on it to protect against script injection attacks.

Reply:
Thanks for your suggestion. Fixed!